src/segtree.rs

```diff
@@ -68,6 +68,48 @@ where
     }
 }
 
+pub struct BitOrOper<S>(Infallible, PhantomData<fn() -> S>);
```

Comment:
This is just a matter of preference, but I prefer `BitwiseOr`.  ("Operation" is obvious in the context, while "bit or" is kinda misleading (are we handling a single bit?))